server/routes/appointments/video-link-booking/handlers/dateAndTime.ts

```diff
@@ -0,0 +1,94 @@
+import { Request, Response } from 'express'
+import { Expose, Transform } from 'class-transformer'
+import { IsEnum, IsNotEmpty, ValidateIf } from 'class-validator'
+import { addMinutes, isValid, startOfToday, subMinutes } from 'date-fns'
+import BookAVideoLinkService from '../../../../services/bookAVideoLinkService'
+import { parseDatePickerDate } from '../../../../utils/datePickerUtils'
+import Validator from '../../../../validators/validator'
+import IsValidDate from '../../../../validators/isValidDate'
+import { YesNo } from '../../../../@types/activities'
+import { dateAtTime, simpleTimeToDate } from '../../../../utils/utils'
+
+export class DateAndTime {
```

Comment:
Not sure if I can think of a better name but this deals with room selection as well, not just date and times.